drivers/gpio/gpio_kb1200.c

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright (c) 2023 ENE Technology Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ene_kb1200_gpio
+
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/gpio/gpio_utils.h>
+#include <zephyr/sys/util_macro.h>
+#include <soc.h>
+
+#define GPIO_REG_BASE ((GPIO_T *)DT_REG_ADDR_BY_NAME(DT_NODELABEL(gpio0x1x), gpio1x))
+#define GPTD_REG_BASE ((GPTD_T *)DT_REG_ADDR_BY_NAME(DT_NODELABEL(gpio0x1x), gptd1x))
+
+struct gpio_kb1200_data {
+	/* gpio_driver_data needs to be first */
+	struct gpio_driver_data common;
+	sys_slist_t cb;
+};
+
+struct gpio_kb1200_config {
+	/* gpio_driver_config needs to be first */
+	struct gpio_driver_config common;
+	/* base address of GPIO port */
+	uint32_t reg;
+	/* GPIO port number */
+	uint8_t port_num;
+};
+
+static void gpio_kb1200_isr(const struct device *dev)
+{
+	const struct gpio_kb1200_config *config = dev->config;
+	struct gpio_kb1200_data *context = dev->data;
+	GPTD_T *gptd_regs = GPTD_REG_BASE;
+	uint32_t pending_flag = gptd_regs->GPTDPFxx[config->port_num];
+
+	gpio_fire_callbacks(&context->cb, dev, pending_flag);
+	gptd_regs->GPTDPFxx[config->port_num] |= pending_flag;
+}
+
+static int kb1200_gpio_pin_configure(const struct device *dev, gpio_pin_t pin, gpio_flags_t flags)
+{
+	const struct gpio_kb1200_config *config = dev->config;
+	GPIO_T *gpio_regs = GPIO_REG_BASE;
+	uint32_t portnum = config->port_num;
+
+	WRITE_BIT(gpio_regs->GPIOFSxx[portnum], pin, 0);
```

Comment:
should probably use `sys_read` / `sys_write` for memory access

Reply:
I referred to other device drivers and few of them use sys_read/sys_write. Is it necessary?